### PR TITLE
Improve test cases

### PR DESCRIPTION
**Pull request by dgr**
Note that this DOESN'T fully close 640. It's just a snapshot of current work.

Please complete and include the following checklist:

- [x] I have read [CONTRIBUTING](https://github.com/clj-commons/etaoin/blob/master/CONTRIBUTING.md) and the [Etaoin Developer Guide](https://github.com/clj-commons/etaoin/blob/master/doc/02-developer-guide.adoc).

- [x] This PR corresponds to an issue that the Etaoin maintainers have agreed to address. 

- [x] This PR contains test(s) to protect against future regressions

- [x] I have updated [CHANGELOG.adoc](https://github.com/clj-commons/etaoin/blob/master/CHANGELOG.adoc) with a description of the addressed issue.

### CODE REVIEW

File: test/etaoin/api_test.clj [hunk 1]
```diff
@@ -799,7 +821,15 @@
...
       (e/add-script *driver* js-url)
-      (e/wait 1)
+      ;; We need to wait for the browser to parse the
+      ;; script. Running "typeof <function_name>" will
+      ;; return "function" if the function is defined (and "undefined"
+      ;; if not).
+      (e/wait-predicate
+       (fn [] (= "function" (e/js-execute *driver* "return typeof injected_func;")))
+       {:timeout 30
+        :internval 0.1
```
Comment (lread) [inline test/etaoin/api_test.clj:831]: Typo: `:internval` should be `:interval`

File: test/etaoin/api_test.clj [hunk 2]
```diff
@@ -647,16 +657,28 @@
 
 (deftest test-switch-window-next
   (let [init-handle (e/get-window-handle *driver*)]
-    (doseq [_ (range 3)]
+    (dotimes [n 3]
```
Comment (lread) [inline test/etaoin/api_test.clj:660]: Right... `dotimes`, nice.

**Review by lread — CHANGES_REQUESTED** [review]
It looks pretty thorough, Dave!
I only noticed one small thing that needs to be corrected.

File: test/etaoin/api_test.clj (hunk 1 above)
Comment (dgr) [reply test/etaoin/api_test.clj]: Fixed and pushed. Thanks.

File: test/etaoin/api_test.clj (hunk 2 above)
Comment (dgr) [reply test/etaoin/api_test.clj:660]: Yea, if you're doing it a specific number of times, it's better to use `dotimes`. It's just semantics, but slightly better semantics.

**Pull request merged**